crates/astria-sequencer/src/fees/mod.rs

```diff
@@ -61,13 +61,35 @@ pub(crate) use state_ext::{
 /// [`tests::get_base_deposit_fee()`].
 const DEPOSIT_BASE_FEE: u128 = 16;
 
+// This module contains the fee handling logic for all actions. All actions must implement the
+// [`FeeHandler`] trait. In addition to this, new actions should provide new, unique read/write
+// methods in [`fees::StateReadExt`] and [`fees::StateWriteExt`] for the fees associated with the
+// action, which take/return a domain type [`<action_name>FeeComponents`]. If there are no fees
+// stored to the state for the given action, it is assumed to be disabled, and the transaction
+// execution should fail. If the action is free, the fees should be explicitly set to zero in the
+// state. See action implementations for examples of how to implement this trait for new actions.
```

Comment:
if this is intended to be a doc comment, it should be at the top of this file, and start with `//!`

Reply:
Do you think this should be a doc comment? I had intentionally left it more as a means of guiding people who are adding actions/fees in the future, since it deals more with the implementation details and figured that probably wouldn't be best for public-facing docs. Happy to make it a doc comment if you think it warrants it, though